crates/frontend/src/components/condition_pills.rs

```diff
@@ -5,6 +5,137 @@ use crate::components::condition_pills::types::ConditionOperator;
 
 use self::types::Condition;
 use leptos::*;
+use wasm_bindgen::JsCast;
+use web_sys::Element;
+
+#[component]
+pub fn condition_expression(
+    #[prop(into)] id: String,
+    #[prop(into)] list_id: String,
+    condition: Condition,
+) -> impl IntoView {
+    let (expand_rs, expand_ws) = create_signal(false);
+
+    let classes = Signal::derive(move || {
+        if expand_rs.get() {
+            (
+                "pointer flex items-center w-max max-w-full rounded-md bg-gray-50 px-2 py-1 text-xs ring-1 ring-inset ring-purple-700/10 shadow-md gap-x-2 overflow-hidden",
+                "font-mono font-semibold context_condition w-full text-wrap word-break-break"
+            )
+        } else {
+            (
+                "pointer flex items-center w-max max-w-[300px] rounded-md bg-gray-50 px-2 py-1 text-xs ring-1 ring-inset ring-purple-700/10 shadow-md gap-x-2 overflow-hidden whitespace-nowrap",
+
+                "font-mono font-semibold context_condition w-full text-ellipsis overflow-hidden whitespace-nowrap"
+            )
+        }
+    });
+
+    let condition = store_value(condition);
+    let id = store_value(id);
+
+    let click_event_handler = window_event_listener(ev::click, move |ev| {
```

Comment:
this is unoptimised, if there are 500 pills at a moment on the page, then it would attach 500 window listeners, this would make things quite very slow

its better, if we would rather attach a single window listener at a parent level to this component, which would store the current clicked pill inside a context, and then the pills can internally check the value of the context to see if that pill is to be expanded or not

Reply:
Or we have the event listener on a class or name so that a single one can be applied to a group of elements?